cmd/bucket-notification-handlers.go

```diff
@@ -170,6 +170,7 @@ func (api objectAPIHandlers) PutBucketNotificationHandler(w http.ResponseWriter,
 
 	rulesMap := config.ToRulesMap()
 	globalNotificationSys.AddRulesMap(bucketName, rulesMap)
+	globalNotificationSys.AddRulesMapRemote(ctx, bucketName, rulesMap)
```

Comment:
Actually, what you need to do not this style, Update() already reloads all bucket metadata remotely. Only fix that is needed is 

```
git diff
diff --git a/cmd/peer-rest-server.go b/cmd/peer-rest-server.go
index 254152ac9..560337367 100644
--- a/cmd/peer-rest-server.go
+++ b/cmd/peer-rest-server.go
@@ -606,6 +606,9 @@ func (s *peerRESTServer) LoadBucketMetadataHandler(w http.ResponseWriter, r *htt
        }
 
        globalBucketMetadataSys.Set(bucketName, meta)
+       if meta.notificationConfig != nil {
+               globalNotificationSys.AddRulesMap(bucketName, meta.notificationConfig.ToRulesMap())
+       }
 }
 
 // ReloadFormatHandler - Reload Format.
```

Reply:
Yeah.. This works :+1